Lib/tarfile.py

```diff
@@ -2349,6 +2349,15 @@ def next(self):
                     raise ReadError(str(e)) from None
             except SubsequentHeaderError as e:
                 raise ReadError(str(e)) from None
+            except Exception as e:
+                try:
+                    import zlib
+                    if isinstance(e, zlib.error):
+                        raise ReadError(f'zlib error: {e}') from None
+                    else:
+                        raise e
+                except ImportError:
+                    raise e
```

Comment:
Observation: this particular `raise` will be pretty noisy since it will be a chained failure. But this is good, it's an unexpected issue for us and we want all the context we can get so that somebody might report a good bug to us.

Reply:
That's what I figured. I could even do `raise Exception('please submit a bug report!') from e` but maybe that is too much.